controllers/datafoundation.go

```diff
@@ -110,6 +119,14 @@ func dfIsReadyToBeRemoved(r *ManagedFusionOfferingReconciler, offering *v1alpha1
 }
 
 func dfReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+	// Check if the required CRDs are installed or not
+	if !doesCRDExistsAtStart {
```

Comment:
Why do we care about existence? what we do care about is if we are watching or not watching specific CRs and we already know that this is based on the CRD map that was loaded in `main.go` and provided to the reconciler.

Also, we should not couple all CRDs under the same umbrella. storagecluster and ocsinitialization are different CRDs and should be treated as such.